File: ballast-core/dotnet/src/application/events/PlayerJoinedGame.cs
using System;
using Ballast.Core.Application.Models;
using Ballast.Core.Messaging;

namespace Ballast.Core.Application.Events
{
    public class PlayerJoinedGameEvent : EventBase, IApplicationEvent  
    {

        public static string GetId() => nameof(PlayerJoinedGameEvent);
        public override string Id => GetId();

        public Guid GameId { get; private set; }
        public PlayerDto Player { get; private set; }

        private PlayerJoinedGameEvent() { } // For model binding / deserialization
        private PlayerJoinedGameEvent(string eventDateIsoString, Guid gameId, PlayerDto player) : base(eventDateIsoString) 
        {
            GameId = gameId;
            Player = player;
        }

        public static PlayerJoinedGameEvent FromPlayerInGame(Guid gameId, PlayerDto player) =>
            new PlayerJoinedGameEvent(
                EventBase.GetDateIsoString(),
                gameId, 
                player
            );

    }
}

File: ballast-core/npm/src/domain/events/player-joined-game.ts
import { EventBase } from "../../messaging/event-base";
import { Game } from "../models/game";
import { Player } from "../models/player";

export class PlayerJoinedGameDomainEvent extends EventBase {

    public static readonly id: string = "PlayerJoinedGameDomainEvent";

    public get id() {
        return PlayerJoinedGameDomainEvent.id;
    }

    public readonly gameId: string; 
    public readonly player: Player;

    private constructor(eventDateIsoString: string, gameId: string, player: Player) {
        super(eventDateIsoString);
        this.gameId = gameId;
        this.player = player;
    }

    public static fromPlayerInGame(game: Game, player: Player) {
        return new PlayerJoinedGameDomainEvent(
            EventBase.getDateIsoString(),
            game.id,
            player
        );
    }

}